docs/connecting.rst

```diff
@@ -0,0 +1,88 @@
+=========================
+Connecting to JanusGraph
+=========================
+
+The library provides a client to connect to JanusGraph running in
+Gremlin Server mode.
+
+For information about Gremlin Server, refer to `JanusGraph docs
+<https://docs.janusgraph.org/latest/server.html>`_.
+
+
+-------------------------
+Build JanusGraph Client
+-------------------------
+
+The client provided to connect to JanusGraph takes care of registering the
+required serializers and deserializers like GeoShape, RelationIdentifier
+and any objects specific to JanusGraph.
+
+Currently implemented for GeoShape for Point and Circle. Needs to be extended.
+
+
+.. code-block:: python
+
+    from janusgraph_python.driver.ClientBuilder import JanusGraphClient
+    # Create JanusGraph connection providing required parameters.
+    connection = JanusGraphClient().connect(host="0.0.0.0", port="8182", graph="g").get_connection()
+    # Create Traversal with JanusGraph connection object
+    g = Graph().traversal().withRemote(connection)
+
+
+-----------------------------------------
+Register Custom Serializer/Deserializer
+-----------------------------------------
+
+JanusGraph Python client provides API to register your custom Serializer/Deserializer
+
+    - Serializer: Serializes a Python object into corresponding JanusGraph object.
+                    Currently implemented for Point,
+                    Circle and
+                    RelationIdentifier
+
+    - Deserializer: Deserializes response from Gremlin Server into corresponding Python objects.
+                    Currently implemented for above mentioned data types
+                    see docs.
+
+
+**NOTE**: It is safe to add Serializer and Deserializer for a particular object together, else we
+might run into unforeseen errors.
+
+**NOTE**: Example considers that you have custom PolygonSerializer defined, Deserializer defined for Polygon and
+a GeoShape for Polygon defined. Since Polygon isn't part of v1 of JanusGraph Python client, it needs to be registered
+separately so as to use Polygon GeoShapes.
+
+.. code-block:: python
+
+    from janusgraph_python.serializer.PolygonSerializer import PolygonSerializer, Polygon
+    from janusgraph_python.serializer.GeoShapeDeserializer import GeoShapeDeserializer
+
+    # All Geoshapes in JanusGraph are identified by `Geoshape`
```

Comment:
How would I modify the comment then? ``All Geoshapes in JanusGraph are identified by namespace `Geoshape` while Relation Identifier by `RelationIdentifier` `` Does this sound good?

Reply:
Sorry, I'm just realizing that I confused the namespace here with the type name. `Geoshape` isn't the namespace, but simply the name of the type. That means however that you need to add the namespace as an argument to the `register_deserializer` method which currently always uses `janusgraph` as the namespace, but users should use their own namespaces for their custom types.

Reply:
Yes, I have a doubt here. So namespace is basically the namespace of Graph DB, and depends on the provider right? So, any custom data types, or new data types introduced by JanusGraph, would always have `janusgraph` as namespace right?

If that is the scenario, then why would we want users to be able to use their own namespace for custom types? Suppose they want to register something like Polygon, or some Data type which is supported by JG, but not by JG-Py, then its namespace will always be `janusgraph` right?

Correct me if my understanding is wrong.

Reply:
If they want to support a datatype that is supported by JanusGraph, but not by the client yet, then yes, they would have to use the `janusgraph` namespace. I only wonder why they would want to do that. If they make the effort to add this support, then why shouldn't they simply summit it as a PR for this library?

The use case for users to add their own serializers should be more to allow them to support their own types. In that case, they probably want to use their own namespace like "myCompanyTypes" to make it easier to debug. Although now that I think more about this, I don't know how useful that is in general if the type is not supported on the server side by JanusGraph. So, we could also omit that functionality for now completely and only add it later in case a user actively asks for it.